common/util/vector_tree.h

```diff
@@ -198,6 +200,172 @@ class _VectorTreeImpl {
   }
 };
 
+namespace vector_tree_internal {
+
+// A wrapper of a sequence container for storing VectorTree nodes that sets
+// correct parent pointer in each inserted node. The reference to the "correct"
+// parent is passed to a constructor.
+//
+// The sole purpose of this class is to function as a children list in
+// VectorTree, both as a storage and as a public interface for tree manipulation
+// (through reference).
+//
+// This class handles parent pointer assignment for all cases where the children
+// list itself is modified. However, it does not clear or otherwise change
+// parent pointer in removed nodes.
+template <class Container>
+class VectorTreeChildrenList
+    : ContainerProxyBase<VectorTreeChildrenList<Container>, Container> {
+  using Base = ContainerProxyBase<VectorTreeChildrenList<Container>, Container>;
+  friend Base;
+
+  using ThisType = VectorTreeChildrenList<Container>;
+
+  using VectorTreeType = typename Container::value_type;
+  friend VectorTreeType;
```

Comment:
Alternative to `friend` relationship, might this helper class be suitable as a private nested class of `VectorTree`?  (Then private access becomes easier.)

Reply:
Good idea, thanks. Implemented & changed the class name to just `ChildrenList`.